django/db/utils.py

```diff
@@ -168,12 +168,21 @@ def ensure_defaults(self, alias):
         except KeyError:
             raise ConnectionDoesNotExist("The connection %s doesn't exist" % alias)
 
-        conn.setdefault('ATOMIC_REQUESTS', False)
-        conn.setdefault('AUTOCOMMIT', True)
+        if hasattr(settings, 'ATOMIC_REQUESTS'):
+            conn.setdefault('ATOMIC_REQUESTS', settings.ATOMIC_REQUESTS)
+        else:
+            conn.setdefault('ATOMIC_REQUESTS', False)
+        if hasattr(settings, 'AUTOCOMMIT'):
+            conn.setdefault('AUTOCOMMIT', settings.AUTOCOMMIT)
+        else:
+            conn.setdefault('AUTOCOMMIT', True)
+        if hasattr(settings, 'CONN_MAX_AGE'):
+            conn.setdefault('CONN_MAX_AGE', settings.CONN_MAX_AGE)
+        else:
+            conn.setdefault('CONN_MAX_AGE', 0)
```

Comment:
These settings should be defined per-`DATABASES` entry and not as top level settings.

e.g.

```python
DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.postgresql',
        'NAME': 'mydatabase',
        'USER': 'mydatabaseuser',
        'PASSWORD': 'mypassword',
        'HOST': '127.0.0.1',
        'PORT': '5432',
        'ATOMIC_REQUESTS': True,
        'AUTOCOMMIT': True,
    }
}
```